### DISCUSSION TITLE
changing the sign of the current function

### DISCUSSION THREAD

**Discussion by akariwrynn** [pybamm-team/PyBaMM#2298] [Q&A]
Hi,
I have been asking some questions regarding current function. The function that I use is "-amplitude * pybamm.sin(2 * np.pi* f * t)". This works perfect for low frequencies but for high frequencies starting from 100Hz, the phase difference becomes pi and I think this should not be happened. As my professor said you have to multiply the current with -1, then the problem will be solved. However, that didn't work either. Do you have any ideas how to make the current negative?

Thanks.

**Comment by akariwrynn** [comment]
It is "amplitude * pybamm.sin(2 * np.pi* f * t)" sorry.

**Reply by rtimms** [reply]
Regarding the sign, we use the convention that positive current corresponds to discharge. For high frequency, you should check that you have enough data points in your solution to capture the output results accurately. Internally the solver will be using adaptive stepping, but will only return the solution at the times in `t_eval`.
